src/main/java/com/microfocus/octane/plugins/configuration/v2/upgrader/UpgraderFromV1ToV2.java

```diff
@@ -0,0 +1,88 @@
+package com.microfocus.octane.plugins.configuration.v2.upgrader;
+
+import com.atlassian.sal.api.pluginsettings.PluginSettings;
+import com.atlassian.sal.api.pluginsettings.PluginSettingsFactory;
+import com.microfocus.octane.plugins.configuration.ConfigurationUtil;
+import com.microfocus.octane.plugins.configuration.LocationParts;
+import com.microfocus.octane.plugins.configuration.v1.ConfigurationCollectionV1;
+import com.microfocus.octane.plugins.configuration.v1.SpaceConfigurationV1;
+import com.microfocus.octane.plugins.configuration.v2.ConfigurationCollectionV2;
+import com.microfocus.octane.plugins.configuration.v2.SpaceConfigurationV2;
+import com.microfocus.octane.plugins.configuration.v2.WorkspaceConfigurationV2;
+import com.microfocus.octane.plugins.tools.JsonHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static com.microfocus.octane.plugins.configuration.ConfigurationManagerConstants.*;
+
+public class UpgraderFromV1ToV2 {
+
+    private static final Logger log = LoggerFactory.getLogger(UpgraderFromV1ToV2.class);
+
+    public static void upgradeConfigurationFromV1ToV2(PluginSettings settings, PluginSettingsFactory pluginSettingsFactory) {
+        ConfigurationCollectionV2 tempConfiguration = null;
```

Comment:
shouldn't you initialize it with new ConfigurationCollectionV2()?

Reply:
no, I want to have new configuration only if the upgrader was successful
otherwise, I'll have a new Config() later, after upgraders